var/spack/repos/builtin/packages/fftw/package.py

```diff
@@ -77,6 +78,60 @@ class Fftw(AutotoolsPackage):
     provides('fftw-api@2', when='@2.1.5')
     provides('fftw-api@3', when='@3:')
 
+    def flag_handler(self, name, flags):
+        spec = self.spec
+        arch = ""
+        simd_x86 = [
+            'sse', 'sse2', 'avx', 'avx2', 'avx512', 'avx-128-fma',
+            'kcvi'
+        ]
+        simd_ppc = ['altivec', 'vsx']
+        simd_arm = ['neon']
+
+        if (spec.satisfies("platform=cray")):
+            # FIXME; It is assumed that cray is x86_64.
+            # If you support arm on cray, you need to fix it.
+            arch  = "x86_64"
+        if (arch != "x86_64" and not spec.satisfies("target=x86_64")):
+            for s in simd_x86:
+                if (spec.satisfies("simd={0}".format(s))):
+                    raise ConflictsInSpecError(
+                        spec,
+                        [(
+                            spec,
+                            spec.architecture.target,
+                            spec.variants['simd'],
+                            'simd=sse,sse2,avx,avx2,avx-128-fma and kcvi'
+                            ' are valid only on x86_64'
+                        )]
+                    )
+        if (arch != "ppc64" and not spec.satisfies("target=ppc64") and
```

Comment:
Can you include `ppc64le` in the target options for Power architectures?

`if (arch not in ('ppc64', 'ppc64le') and not any(spec.satisfies(cond) for cond in ('target=ppc64', 'target=ppc64le', 'target=power7')):`